pipeline/README.md

```diff
@@ -333,15 +282,70 @@ useSrlNom   true
 Note that individuals have their own configuration options -- see the documentation
 for individual components for details.
 
-
-
 ### Changing the logging settings
 
 This project uses slf4j's log4j libraries.  You can change the
 settings by creating a log4j.properties file and adding the directory
 containing that file to the classpath.
 
-### Frequently Asked Questions (FAQs)
+## Using pipeline webserver 
+
+Often a convenient model of using the pipeline server is, running the server (which includes all the annotators) on a 
+big machine (=big memory) and sending calls to the server with clients. Here we first introduce the details of 
+the server and later we will delineate the clients.
+ 
+The server supports post and get requests to obtain  annotation for a requested text, with desired views. 
+In order to run the webserver with default settings (port = 8080), do: 
+
+```shell
+pipeline/scripts/runWebserver.sh
+```
+
+The following arguments are supported:
+```shell
+usage: pipeline/scripts/runWebserver.sh [-h] [--port PORT]
+
+optional arguments:
+  -h, --help             show this help message and exit
+  --port PORT, -P PORT   Port to run the webserver.
+```
+
+Here are the available APIs: 
+
+
+| API                    | Address      | Supported request type | Parameters                                                                   | Example                                                          |
+|------------------------|--------------|------------------------|------------------------------------------------------------------------------|------------------------------------------------------------------|
+| Annotating text        | `/annotate`  | POST/GET               | `text`: the target raw text ; `views`: views to be added, separated by comma | `/annotate?text="This is sample text"&views=POS,NER_CONLL` |
+| Getting existing views | `/viewNames` | POST/GET               | N/A                                                                          | `/viewNames`                                                     |
+
+Note that the current webserver is a very basic one and very small sophistications. It does not support any parallel  annotations of single request, or parallel processing of multiple requests. Such extensions are in our TODO-list for future. 
```

Comment:
Note that the current web server is very basic.  It does not support parallel processing within a single request, nor across multiple requests.